app/controllers/users_controller.rb

```diff
@@ -0,0 +1,82 @@
+class UsersController < ApplicationController
+  include UserSessionsHelper
+  before_action :set_user, only: %i[show edit]
+
+  # GET /users
+  # GET /users.json
+  def index
+    @users = User.all
+    @user = User.new
+  end
+
+  # GET /users/1
+  # GET /users/1.json
+  def show
+    user = User.find(params[:id]) # Encontramos usuario
+    @user_events = user.events unless user.nil? # Toda la lsita de eventos del usuario show
+    @user_attended = user.attended_events
+    @upcoming_events = @user_attended.upcoming_events
+    @past_events = @user_attended.past_events
+  end
```

Comment:
For the `user` lets use `current_user` for example `current_user.events`

Reply:
current_user is a built-in helper method we have created to help us identify the User that is in the current session (or currently logged in). If we were to use this, whenever we show a user, the issue would be that it would only show us the event pertaining to the currently signed in User. However, what we wish to achieve is to show every event with the information of the user that created SUCH events.  If we were to use current_user.events then it would only show the creator of the event if they're in the current session (currently signed in). Could you please help us understand why you wish us to implement this change?

Reply:
Here we want to get 
- [ ] lists events created by user
- [ ]  lists events previously attended by user using event model scope
- [ ] lists upcoming events previously attended by user using event model scope
Thats why `current_user`  helper method is needed here